File: modulewh/clientwh/src/main/resources/static/js/service/materialbaselinedetailService.js


/**
 * Service for Materialbaselinedetail
 **/

define(['require', 'angular'], function (require, angular) {
app.aService(clientwh.prefix + 'materialbaselinedetailService', function($http, $rootScope) {
	
	// Create.
	this.create = function(materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/create';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
	
	// Update Lock.
	this.updateLock = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateLock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update Unlock.
	this.updateUnlock = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateUnlock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update.
	this.update = function(id, materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/update/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
	
	// Update With Lock.
	this.updateWithLock = function(id, materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateWithLock/' + id;
		var request = {
				 method: 'PUT',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
	
	// Update For Delete.
	this.updateForDelete = function(id, version) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateForDelete/' + id + '/' + version;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Update For Delete With Lock.
	this.updateForDeleteWithLock = function(id, version) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateForDeleteWithLock/' + id + '/' + version;
		var request = {
				 method: 'PUT',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Delete.
	this.delete = function(id) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/delete/' + id;
		var request = {
				 method: 'DELETE',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Get by Id.
	this.getById = function(materialbaselinedetailId) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/getById/' + materialbaselinedetailId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// List for page and filter.
	this.listWithCriteriasByPage = function(criterias, pageNo, pageSize, sorts) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/listWithCriteriasByPage?' + 'page=' + pageNo + '&size=' + pageSize;
		if(typeof(sorts) !== 'undefined' && sorts.length > 0) {
			angular.forEach(sorts, function(sort) {
				serverUrl += '&' + sort;
			});
		}
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: criterias
				}
		return $http(request);
	}
	
	// List for page and filter.
	this.listWithCriteriasByIdmaterialbaselineAndPage = function(idmaterialbaseline, criterias, pageNo, pageSize, sorts) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/listWithCriteriasByIdmaterialbaselineAndPage/' + idmaterialbaseline + '?page=' + pageNo + '&size=' + pageSize;
		if(typeof(sorts) !== 'undefined' && sorts.length > 0) {
			angular.forEach(sorts, function(sort) {
				serverUrl += '&' + sort;
			});
		}
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: criterias
				}
		return $http(request);
	}
	
	// List all material idref = 0 for select.
	this.listAllForSelect = function(){
		var serverUrl = clientwh.serverUrl + '/material/listAllForSelect';
		var request = {
				method: 'GET',
				url: serverUrl
		}
		return $http(request);
	}
	
	// List all supplier for select.
	this.listAllForSelect = function(){
		var serverUrl = clientwh.serverUrl + '/supplier/listAllForSelect';
		var request = {
				method: 'GET',
				url: serverUrl
		}
		return $http(request);
	}
	
	// List of amount for select.
	this.sumAmount = function(idmaterialbaseline) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/sumAmount/' + idmaterialbaseline;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}

	// Get All by Id.
	this.getAllById = function(materialbaselineId) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/getAllById/' + materialbaselineId;
		var request = {
				 method: 'GET',
				 url: serverUrl
				}
		return $http(request);
	}
	
	// Create.
	this.createWithSelectDetail = function(materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/createWithSelectDetail/';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
	
	// Update price.
	this.materialbaselineUpdatePrice = function(materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updatePrice/';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
	
	// Update quantity.
	this.materialbaselineUpdateQuantity = function(materialbaselinedetail) {
		var serverUrl = clientwh.serverUrl + '/materialbaselinedetail/updateQuantity/';
		var request = {
				 method: 'POST',
				 url: serverUrl,
				 data: materialbaselinedetail
				}
		return $http(request);
	}
});

});

File: modulewh/serverwh/src/main/java/com/redsun/server/wh/service/MaterialbaselinedetailService.java

package com.redsun.server.wh.service;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.query.Param;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.redsun.server.wh.model.Materialbaselinedetail;
import com.redsun.server.wh.model.common.SearchCriteria;

public interface MaterialbaselinedetailService {

	/**
	 * Saves (create or update) the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 */
	Materialbaselinedetail save(Materialbaselinedetail materialbaselinedetail);

	/**
	 * Create the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 */
	Materialbaselinedetail create(Materialbaselinedetail materialbaselinedetail) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateLock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @param idlock
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateUnlock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialbaselinedetail update(Integer id, Materialbaselinedetail materialbaselinedetail) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialbaselinedetail updateWithLock(Integer id, Materialbaselinedetail materialbaselinedetail) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialbaselinedetail updateForDelete(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialbaselinedetail updateForDeleteWithLock(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	
	/**
	 * Deletes the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialbaselinedetail
	 * @return true if found and deleted, false if not found
	 */
	void delete(Materialbaselinedetail materialbaselinedetail);

	/**
	 * Deletes the entity by its Primary Key <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return true if found and deleted, false if not found
	 */
	void deleteById(Integer id);

	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	Materialbaselinedetail getById(Integer id);

	/**
	 * Loads ALL the entities (use with caution)
	 * @return
	 */
	List<Materialbaselinedetail> listAll();

	/**
	 * Count all the occurrences
	 * @return
	 */
	long countAll();

	/**
	 * Check exist
	 * @return
	 */
	boolean isExist(Integer id);

	/**
	 * Search by a criteria.
	 * @return
	 */
	List<Materialbaselinedetail> listWithCritera(SearchCriteria searchCriteria);

	/**
	 * Search by multiple criteria.
	 * @return
	 */
	List<Materialbaselinedetail> listWithCriteras(List<SearchCriteria> searchCriterias);

	/**
	 * Loads ALL the entities by page (use with caution)
	 * @return
	 */
	Page<Materialbaselinedetail> listAllByPage(Pageable pageable);

	/**
	 * Search by a criteria by page.
	 * @return
	 */
	Page<Materialbaselinedetail> listWithCriteraByPage(SearchCriteria searchCriteria, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Materialbaselinedetail> listWithCriterasByPage(List<SearchCriteria> searchCriterias, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Map<String, Object>> listWithCriteriasByIdmaterialbaselineAndPage(Integer idmaterialbaseline, List<SearchCriteria> searchCriterias, Pageable pageable);
	
	Double sumAmount(@Param("idmaterialbaseline") Integer idmaterialbaseline);
	
	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	List<Map<String, Object>> getAllById(Integer id);

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id,price
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	void updatePrice(Integer id, Double price) throws JsonParseException, JsonMappingException, IOException ;
	
	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id,quantity
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	void updateQuantity(Integer id, Integer quantity) throws JsonParseException, JsonMappingException, IOException ;
}
